# No Wildcard Imports

## Rule Details

This rule enforces that no sx props are used with `@primer/react`.

ğŸ‘ Examples of **incorrect** code for this rule

```jsx
import {Button} from '@primer/react'

function ExampleComponent() {
  return (
    <Button
      sx={
        {
          /* ... */
        }
      }
    />
  )
}
```

ğŸ‘ Examples of **correct** code for this rule:

```jsx
import {Button} from '@primer/react'

function ExampleComponent() {
  return <Button className="..." />
}
```
